### DISCUSSION TITLE
Visitor authentication via SAML/SSO

### DISCUSSION THREAD

**Discussion by Kaustubh804975** [GitbookIO/community#247] [Feature Requests]
### What's your feature/idea?

Hi Team,

I would like to share a Idea or it can be a kind of enhancement to a feature which Gitbook has, which is **Publish with visitor authentication**
Here we are able to see that is option given to **Choose the backend used to authenticate visitors** and by default its Custom.
We also have SSO & SAML for authentication.
![image](https://github.com/GitbookIO/community/assets/135026988/08fb1706-56a9-4cbc-85d0-4d8c681eeaed)

My request would be to enhance/ improve **Choose the backend used to authenticate visitors** feature, and as gitbook has SAML authentication, It would be great if we are able to opt for SAML from **Choose the backend used to authenticate visitors** and share publish the document with visitors authentication and user who already has SAML access will click on the shared URL and will be authenticate via SAML.

### What's your desired outcome?

If would be really helpful if we have to share any documents which can be customized and shared with whole Organization without using JWT or hosting a server for this. 

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by addisonschultz** [comment]
Hey @Kaustubh804975! Thanks for the request - we definitely want to improve this in the future, by adding some common providers like the one(s) you mentioned. I don't have a timeline on this at the moment, but just wanted to let you know it's on our radar!

**Comment by taszap** [comment]
Related to #306
